addons/project/static/src/views/project_task_kanban/project_task_kanban_model.js

```diff
@@ -1,13 +1,88 @@
 /** @odoo-module */
 
 import { RelationalModel } from "@web/model/relational_model/relational_model";
+import { Record } from "@web/model/relational_model/record";
+import { makeActiveField } from "@web/model/relational_model/utils";
 
 export class ProjectTaskKanbanDynamicGroupList extends RelationalModel.DynamicGroupList {
     get isGroupedByStage() {
         return !!this.groupByField && this.groupByField.name === "stage_id";
     }
+
+    get isGroupedByPersonalStages() {
+        return !!this.groupByField && this.groupByField.name === "personal_stage_type_ids";
+    }
+
+    async createGroup(groupName, foldField) {
+        if (this.isGroupedByPersonalStages) {
+            return this.model.mutex.exec(async () => {
+                this.creatingPersonnalStage = true;
+                await this._createGroup(groupName, foldField);
+                delete this.creatingPersonnalStage;
+            });
+        }
+        return super.createGroup(...arguments);
+    }
+
+    async _unlinkGroups(groups) {
+        if (this.isGroupedByPersonalStages) {
+            const groupResIds = groups.map((g) => g.value);
+            return this.model.orm.call("project.task.type", "remove_personal_stage", groupResIds);
+        }
+        return super._deleteGroups(...arguments);
+    }
+}
+
+export class ProjectTaskRecord extends Record {
+    setup() {
+        super.setup(...arguments);
+        this.displaySubtasks = false;
+    }
+
+    async toggleSubtasksList() {
+        const { display_name, project_id, state, user_ids } = this.config.fields;
+        const activeField = makeActiveField({ onChange: true });
+        activeField.related = {
+            activeFields: {
+                display_name: makeActiveField(),
+                state: makeActiveField(),
+                user_ids: makeActiveField(),
+                project_id: makeActiveField(),
+            },
+            fields: {
+                display_name,
+                project_id,
+                state,
+                user_ids,
+            },
+        };
+        await this._load({
+            activeFields: { ...this.config.activeFields, child_ids: activeField },
+        });
+        this.displaySubtasks = !this.displaySubtasks;
+    }
+
+    async _applyChanges(changes, serverChanges = {}) {
+        const value = changes.personal_stage_type_ids;
+        if (Array.isArray(value)) {
+            delete changes.personal_stage_type_ids;
+            changes.personal_stage_type_id = value;
+        }
+        await super._applyChanges(changes, serverChanges);
+    }
+
+    get context() {
+        const context = super.context;
+        const value = context.default_personal_stage_type_ids;
+        if (Array.isArray(value)) {
+            context.default_personal_stage_type_id = value[0];
+            delete context.default_personal_stage_type_ids;
+        }
+        return context;
+    }
```

Comment:
I think it is no longer needed since the refactoring made by Hugo, right? 🧐

Reply:
:+1: